### PR TITLE
boards: opta: enable ADC, RS485 and ethernet reorganization

### PR DESCRIPTION
**Pull request by fogzot**
This set of changes enables the RS485 hardware connected to usart3 and provides some overlays that allow for easily running the modbus rtu_client and rtu_server samples on Opta.

A valid device tree configuration is provided for the ADCs of the 8 input channels and the sample adc_dt works out of the box. Obviously this is not the only possible configuration but it provides a good template for further customization without the need to lookup the ADC GPIOs and connections in the schematics.

Also, the `status = "okay"` was removed from ethernet configuration in the device tree to avoid always committing 16KB of RAM for the network buffers. As shown in the new overlay for the telnet sample it is sufficient to add it back to get a fully working network connection.

### CODE REVIEW

**Comment by jukkar** [comment]
As this PR seems to be having three components that are being changed/reorganized, I suggest separating them to three commits for easier review and maintenance.

**Comment by fogzot** [comment]
> As this PR seems to be having three components that are being changed/reorganized, I suggest separating them to three commits for easier review and maintenance.

I separated the changes in three commits but now I have a doubt: do you mean a single pull request with 3 commits or 3 different pull requests?

**Comment by jukkar** [comment]
> I separated the changes in three commits but now I have a doubt: do you mean a single pull request with 3 commits

Yes, this is what I meant. No need to have three PRs.

**Comment by fogzot** [comment]
Done. The chnages have been split into 3 commits:
1) ethernet reorganization to avoid consuming RAM when not needed
2) RS485 support
3) ADC support

**Review by pillo79 — COMMENTED** [review]
Thanks for the split, looks a lot cleaner now and mostly OK to me. :+1: 
However, the Ethernet commit still contains unrelated DTS file changes (SDRAM, Flash, UART, USB). I think a separate commit for these with a little background on the changes would be helpful.

**Comment by fogzot** [comment]
> However, the Ethernet commit still contains unrelated DTS file changes (SDRAM, Flash, UART, USB). I think a separate commit for these with a little background on the changes would be helpful.

@pillo79 done. Note that the USB change in the first commit (ethernet) does not change anything: it's just a comment clarifying that the USB is assigned to M7 by default to have a working console in the common case.

**Review by pillo79 — DISMISSED** [review]
Thanks, a lot cleaner now. LGTM!

File: boards/arduino/opta/arduino_opta-common.dtsi [hunk 1]
```diff
@@ -58,6 +58,14 @@
 		};
 	};
 
+	sdram2: sdram@d0000000 {
```
Comment (facchinm) [inline boards/arduino/opta/arduino_opta-common.dtsi:61]: Unluckily, Opta doesn't have 32MB of SDRAM :sweat_smile: 
Please remove the entry.

**Review by facchinm — CHANGES_REQUESTED** [review]

File: boards/arduino/opta/arduino_opta-common.dtsi (hunk 1 above)
Comment (fogzot) [reply boards/arduino/opta/arduino_opta-common.dtsi]: My fault, I was too fast with a copy&paste when I setup the eth part and then never asked myself why that entry was there. Removed.

File: boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts [hunk 2]
```diff
@@ -23,22 +23,6 @@
...
-		status = "okay";
-	};
-
-	sdram2: sdram@d0000000 {
-		compatible = "zephyr,memory-region", "mmio-sram";
-		device_type = "memory";
-		reg = <0xd0000000 DT_SIZE_M(32)>;
-		zephyr,memory-region = "SDRAM2";
-		zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM) )>;
-	};
```
Comment (pillo79) [inline boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts:41]: This part of the change goes to the "device tree cleanup" commit. Also update that commit's description as it still refers to the SDRAM move.

**Review by pillo79 — COMMENTED** [review]
Seriously nitpicking here, but since it is so clean now it's easy to spot the issues :slightly_smiling_face: . Otherwise LGTM!

Note: please do not rebase a PR unless it is required to fix merge issues (Github will warn you if there are any!). Rebasing makes life more difficult for reviewers as the Github "Compare" button becomes useless.

**Review by facchinm — DISMISSED** [review]

File: boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts (hunk 2 above)
Comment (fogzot) [reply boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts:41]: Sorry for the rebase, other projects I work with have the exact opposite policy (always rebase).

Anyway, I think now everything is as required.

**Review by pillo79 — DISMISSED** [review]
Thanks for bearing with us!

**Comment by fogzot** [comment]
@facchinm @pillo79 I tried running the failing test on my local machine in every possible way and it never fails. Do you have any suggestions about how to replicate the CI environment to understand what's happening?

**Comment by pillo79** [comment]
@fogzot I was always lucky that those issues usually fixed themselves with a rebase on the latest main, but it wasn't the case here. :slightly_frowning_face: 
You can download the docker image used by CI with [these instructions](https://github.com/zephyrproject-rtos/docker-image), so you could try running the test there. More info on the actual full CI setup should be in the `.github/` folder in the repo, but not sure how much can be replicated locally.

File: boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts [hunk 3]
```diff
@@ -124,13 +112,12 @@ zephyr_udc0: &usbotg_fs {
 		&eth_txd0_pg13
 	>;
 	pinctrl-names = "default";
-	status = "okay";
```
[the remark this answers is not in the document]
Comment (fogzot) [reply boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts:127]: A third option is to add an overlay to the test to enable the ethernet. Is this acceptable?

[the remark this answers is not in the document]
Comment (pillo79) [reply boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts:127]: I vote for a working Ethernet by default. No other board has an overlay on that test, it's a hint I guess :slightly_smiling_face: 
Go ahead with the rebase.

[the remark this answers is not in the document]
Comment (fogzot) [reply boards/arduino/opta/arduino_opta_stm32h747xx_m7.dts]: Done. Also rephrased a bit the commit message.

**Review by pillo79 — APPROVED** [review]
At last! :partying_face: Thanks for this work!

**Review by jukkar — APPROVED** [review]

**Pull request merged**